lib/generators/hyrax/work/work_generator.rb

```diff
@@ -15,6 +18,25 @@ def banner
     end
   end
 
+  # rubocop:disable Rails/Exit
+  # rubocop:disable Style/GuardClause
+  def check_inflection_behavior
+    # Don't short-circuit if destroying the work type, assuming it was
+    # generated before this new logic was added
+    return if revoking?
+
+    # Mimic logic from Rails:
+    # https://github.com/rails/rails/blob/92703a9ea5d8b96f30e0b706b801c9185ef14f0e/railties/lib/rails/generators/model_helpers.rb#L16
+    # To prevent problems with types named, e.g., "Atlas" or "Lens"
+    #
+    if file_name == plural_file_name && file_name.singularize != file_name
+      say_status("error", format(PLURAL_MODEL_NAME_WARN_MESSAGE, file_name, file_name.singularize), :red)
+      exit
```

Comment:
You want to unilaterally `exit`?  When this engine (as seems inevitable in our stack) is called through a chain of 3 other engines, you want to **prevent** them from being able to swallow an exception and keep on going?